Implement get_file_bcontents() method

GouthamOfTheNP opened the pull request:
Add method to retrieve raw byte contents of a file in reference to the feature requested by issue #85852

##### SUMMARY

<!--- Describe the change below, including rationale and design decisions -->
Added a new method that mirrors the DataLoader.get_text_file_contents() method named DataLoader.get_text_file_bcontents() that returns a tagged copy of the raw bytes of a file, as requested by a user in the aforementioned issue, in order to improve the API user experience.
<!--- Add "Fixes #1234" if there is a corresponding issue -->

##### ISSUE TYPE

- Feature Pull Request

GouthamOfTheNP committed:
Implement get_text_file_bcomponents method

Add method to retrieve raw byte contents of a file in reference to the feature requested by issue #85852

webknjaz commented:
Changes like this tend to require integration tests and change log fragments.

GouthamOfTheNP commented:
Is this a pass or a fail based on the pass rate (93.73%)?

GouthamOfTheNP committed:
Rename get_text_file_bcomponents to get_file_bcomponents

GouthamOfTheNP committed:
Rename method get_file_bcomponents to get_file_bcontents

webknjaz commented:
> Is this a pass or a fail based on the pass rate (93.73%)?

Just look at the checks widget in PRs. If things are green, the azp side stats don't matter in the scope of this PR.

Akasurde commented on lib/ansible/parsing/dataloader.py:
```diff
@@ -191,6 +191,17 @@ def get_text_file_contents(self, file_name: str, encoding: str | None = None) ->
 
         return AnsibleTagHelper.tag_copy(bytes_content, str_content)
 
+    def get_file_bcontents(self, file_name: str) -> bytes:
```
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

GouthamOfTheNP committed:
Implement get_file_bcomponents() method in dataloader

Added the `get_file_bcomponents()` method to the dataloader, which returns raw byte contents of a file while preserving `Origin` and `SourceWasEncrypted` tags without decoding.

felixfontein commented on lib/ansible/parsing/dataloader.py:
```diff
@@ -191,6 +191,17 @@ def get_text_file_contents(self, file_name: str, encoding: str | None = None) ->
+        if not source_was_plaintext:
+            bytes_content = SourceWasEncrypted().tag(bytes_content)
+
+        return AnsibleTagHelper.tag_copy(bytes_content, bytes_content)
```
Is `tag_copy` needed here? After all, both arguments to it are identical.

GouthamOfTheNP committed:
Simplify return value in dataloader.py

Removed redundant code (AnsibleTagHelper.tag_copy() to just bytes)
